rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll

```diff
@@ -135,7 +147,9 @@ module Impl {
     predicate isCaptured() { this.getAnAccess().isCapture() }
 
     /** Gets the parameter that introduces this variable, if any. */
-    Param getParameter() { parameterDeclInScope(result, this, _) }
+    ParamBase getParameter() {
+      result = this.getSelfParam() or result = getAVariablePatAncestor(this).getParentNode()
```

Comment:
Relying on `getParentNode()` is generally not good, as it is basically an untyped interface. So I would prefer if we could revert it back to using `parameterDeclInScope`.